lib/exe/css.php

```diff
@@ -344,6 +360,37 @@ function css_fixreplacementurls($replacements, $location) {
 }
 
 /**
+ * Wrapper for the files, content and mediatype for the event CSS_STYLES_INCLUDED
+ *
+ * @param string $mediatype type ofthe current media files/content set
+ * @param array $files set of files that define the current mediatype
+ * @return array
+ */
+function css_filewrapper($mediatype, $files=array()){
+    return array(
+            'files'                 => $files,
+	        'mediatype'             => $mediatype,
+	        'encapsulate'           => in_array($mediatype, array('screen', 'print')),
```

Comment:
Why allow any kind of media type and then just put them all in one bucket?
This goes against what a media type is. The list here should include all valid media types except `all`. And `all` should be the only one allowed without a wrapper.

On the other hand, it seems to be deliberate to allow any random media, but I don't understand why. Could it be you're only piggybacking media types for something unrelated? If yes, then you should create a completely new functionality for this.